### PR TITLE
Make the Concent Service optional from Provider's POV

### PR DESCRIPTION
**Pull request by shadeofblue**
adds two RPC endpoints:

`golem.concent.required_as_provider`
and
`golem.concent.required_as_provider.turn`

(analogous to `golem.concent.switch`)

that can be used to make the Concent Service either required (default) or optional from Provider's POV 

if required is on, Provider won't accept any tasks that have `concent_enabled=False`
if required is off on the other hand, Provider will accept both the tasks that have `concent_enabled=True` and those where `concent_enabled` is `False`

### CODE REVIEW

File: golem/task/tasksession.py [hunk 1]
```diff
@@ -464,8 +465,13 @@ def _cannot_compute(reason):
...
 
-        if self.concent_service.enabled and not msg.concent_enabled:
-            # Provider requires concent if it's enabed locally
+        if (
+                self.concent_service.enabled
+                and self.concent_service.required_as_provider
+                and not msg.concent_enabled
+        ):
+            # Provider requires concent
+            # if it's enabed locally and marked as required
```
Comment (kmazurek) [inline golem/task/tasksession.py]: typo: enabed -> enabled

File: golem/network/concent/soft_switch.py [hunk 2]
```diff
@@ -9,24 +9,55 @@ class SoftSwitch(enum.Enum):
...
+@rpc_utils.expose('golem.concent.required_as_provider')
+def is_required_as_provider() -> bool:
+    """
+    Verify if the Concent is required for tasks accepted for computation
+    as a Provider
+    """
+    return _is_on(REQUIRED_KEY, default=SoftSwitch.ON)
+
+
+@rpc_utils.expose('golem.concent.required_as_provider.turn')
```
Comment (kmazurek) [inline golem/network/concent/soft_switch.py:58]: This name is... confusing. But I understand where it's coming from. Still, as discussed, I think we could change this to a single RPC endpoint per setting (both for getting and updating its value, depending on whether a parameter is passed).

File: tests/golem/network/concent/test_soft_switch.py [hunk 3]
```diff
@@ -2,15 +2,50 @@
 from golem.tools.testwithdatabase import TestWithDatabase
 
 
-class TestContentSoftSwitch(TestWithDatabase):
+class ContentSwitchTestMixin:
```
Comment (kmazurek) [inline tests/golem/network/concent/test_soft_switch.py]: I believe you meant `ConcentSwitchTestMixin`

File: tests/golem/network/concent/test_soft_switch.py [hunk 4]
```diff
@@ -2,15 +2,50 @@
...
     def test_turn_off(self):
-        soft_switch.turn(True)
-        soft_switch.turn(False)
-        self.assertFalse(soft_switch.is_on())
+        self._turn(True)
+        self._turn(False)
+        self.assertFalse(self._is_on())
+
+
+class TestContentSoftSwitch(ContentSwitchTestMixin, TestWithDatabase):
```
Comment (kmazurek) [inline tests/golem/network/concent/test_soft_switch.py]: ditto, Content -> Concent

File: tests/golem/network/concent/test_soft_switch.py [hunk 5]
```diff
@@ -2,15 +2,50 @@
...
+        return False
+
+    def _turn(self, on: bool):
+        return soft_switch.concent_turn(on)
+
+    def _is_on(self):
+        return soft_switch.concent_is_on()
+
+
+class TestContentRequiredAsProvider(ContentSwitchTestMixin, TestWithDatabase):
```
Comment (kmazurek) [inline tests/golem/network/concent/test_soft_switch.py]: ditto

File: tests/golem/network/concent/test_soft_switch.py (hunk 3 above)
Comment (shadeofblue) [reply tests/golem/network/concent/test_soft_switch.py]: ugh :)

**Review by kmazurek — APPROVED** [review]
LGTM

**Pull request merged**